BlueprintUICommonControls/Sources/AccessibilityFocus.swift

```diff
@@ -0,0 +1,117 @@
+import BlueprintUI
+import UIKit
+
+/// Enables VoiceOver focus to jump to the wrapped element
+/// either on appearance or via a trigger that can be manually fired.
+public struct AccessibilityFocus: Element {
+
+    /// The element that will have the focus.
+    public var wrapping: Element
+
+    /// The type of accessibility notification that will be triggered.
+    public var notificationType: UIAccessibility.Notification
+
+    /// A object that can be held on to by the caller to manually trigger a focus.
+    public var trigger: Trigger
+
+    // MARK: Element
+
+    public var content: ElementContent {
+        ElementContent(child: wrapping)
+    }
+
+    public func backingViewDescription(with context: ViewDescriptionContext) -> ViewDescription? {
+        View.describe { config in
+            config[\.isAccessibilityElement] = false
+            config.apply { view in
+                view.apply(model: self)
+            }
+        }
+    }
+}
+
+// MARK: Backing view
+
+extension AccessibilityFocus {
+    private final class View: UIView {
+
+        func apply(model: AccessibilityFocus) {
+            model.trigger.action = { [weak self] in
+                guard let self = self else { return }
+                UIAccessibility.post(notification: model.notificationType, argument: self)
+            }
+
+            /// We need to wait a moment before firing the focus, to ensure that
+            /// we are not competing with UIAccessibility for the focus.
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
+                model.trigger.fireImmediately()
```

Comment:
Fwiw there's no guarantee that this is in a view hierarchy or window hierarchy when this is invoked. Should this be done in `viewDidMoveToWindow` or something? What is the use case here? It feels fragile, especially in list views - where the cells, which are recycled, remain in the list forever – but are not always visible.

Reply:
The goal here was to trigger the accessibility focus on any element that just appeared on the screen or if there's any one element we would like to take automatic accessibility focus when a new screen loads. In the scenario you describe, do we set the model as a property on the `View` class through the `backingViewDescription` and execute this code inside `didMoveToWindow()`?

Reply:
Something like that... or as mentioned in another comment, hook this into the `onAppear` machinery in Blueprint? Not 100% sure.

Reply:
Yeah, Kyle is correct, you can't rely on this, and we don't support it in "normal" focus for that reason. To focus on appear you'd probably want a construction like this:

```swift
myElement
  .accessibilityFocus(trigger: trigger)
  .onAppear {
    trigger.fire()
  }
```